### PR TITLE
Run commit hooks using `pre-commit`

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <hkawamura0130@gmail.com>

## Related Issues/PRs

 #xxx

## What changes are proposed in this pull request?

Run commit hooks using https://pre-commit.com. We can easily add more hooks if necessary in the future (e.g. a hook to enforce a new line at EOF or eliminate trailing whitespace).

## How is this patch tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [x] Manual tests (describe details, including test results, below)

```diff
> git diff
diff --git a/mlflow/version.py b/mlflow/version.py
index 0aa716cd5..6e564ce85 100644
--- a/mlflow/version.py
+++ b/mlflow/version.py
@@ -5,5 +5,9 @@ import re
 VERSION = "2.0.0.dev0"
 
 
+def foo(a,   b):
+    pass
+
+
 def is_release_version():
     return bool(re.match(r"^\d+\.\d+\.\d+$", VERSION))

```

```
> git commit -a -m "test"
black....................................................................Failed
- hook id: black
- exit code: 1

would reformat mlflow/version.py

Oh no! 💥 💔 💥
1 file would be reformatted, 704 files would be left unchanged.

pylint...................................................................Failed
- hook id: pylint
- exit code: 4

************* Module mlflow.version
mlflow/version.py:8:8: W0613: Unused argument 'a' (unused-argument)
mlflow/version.py:8:13: W0613: Unused argument 'b' (unused-argument)

-------------------------------------------------------------------
Your code has been rated at 6.67/10 (previous run: 10.00/10, -3.33)



rstcheck.............................................(no files to check)Skipped
```

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/version.py [hunk 1]
```diff
@@ -5,5 +5,9 @@
 VERSION = "2.0.0.dev0"
 
 
+def foo(a,   b):
+    pass
+
+
```
Comment (harupy) [inline mlflow/version.py:11]: Test

Comment (harupy) [inline mlflow/version.py:11]:
```suggestion
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -287,7 +287,7 @@ commits are signed-off and runs `black --check` and `pylint` to ensure the
 code will pass the lint check for python. You can enable it by running:
 
 ```bash
-git config core.hooksPath hooks
+pre-commit install -t pre-commit -t prepare-commit-msg
```
Comment (BenWilson2) [inline CONTRIBUTING.md:290]: Should we update the dev setup script to force-install this during script execution?

**Review by BenWilson2 — APPROVED** [review]
LGTM!

File: CONTRIBUTING.md (hunk 2 above)
Comment (harupy) [reply CONTRIBUTING.md:290]: Updated!

**Pull request merged**